test/s3select_test_json_parser.cpp

```diff
@@ -0,0 +1,561 @@
+wKICAgICAgICAgICAgICAgICAgICAgICAgIkdsb3NzTGlzdCI6IHsKICAgICAgICAgICAgICAgICJHbG9zc0VudHJ5IjogewogICAgICAgICAgICAgICAgICAgICJJRCI6ICJTR01MIiwKICAgICAgI\
+CAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICJTb3J0QXMiOiAiU0dNTCIsCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAiR2xvc3NUZXJtIjogIlN0YW5kYXJk\
+IEdlbmVyYWxpemVkIE1hcmt1cCBMYW5ndWFnZSIsCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAiQWNyb255bSI6ICJTR01MIiwKICAgICAgICAgICAgICAgICAgICAgIC\
+AgICAgICAgICAgICAgICAgICJBYmJyZXYiOiAiSVNPIDg4Nzk6MTk4NiIsCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAiR2xvc3NEZWYiOiB7CiAgICAgICAgICAgICAg\
+ICAgICAgICAgICJwYXJhIjogIkEgbWV0YS1tYXJrdXAgbGFuZ3VhZ2UsIHVzZWQgdG8gY3JlYXRlIG1hcmt1cCBsYW5ndWFnZXMgc3VjaCBhcyBEb2NCb29rLiIsCiAgICAgICAgICAgICAgICAgIC\
+AgICAgICAgICAgICAgICAgICAgICAgICAgICAgICJHbG9zc1NlZUFsc28iOiBbIkdNTCIsICJYTUwiXSwKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgInBv\
+c3RhcnJheSI6IHsKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICJhIjoxMTEsCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIC\
+AgICAgICAgICAgICAgICAgICAgICAgICAiYiI6MjIyCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIH0KICAgICAgICAgICAgICAgICAgICB9LAogICAgICAg\
+ICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIkdsb3NzU2VlIjogIm1hcmt1cCIKICAgICAgICAgICAgICAgIH0KICAgICAgICAgICAgfQogICAgICAgIH0KICAgIH0KfQo="
+
+std::string run_sax(const char * in)
+{
+	JsonParserHandler handler;
+
+	handler.key_value_criteria = true;
+
+	int status = handler.process_json_buffer(base64_decode(std::string(in)).data(), strlen(in));
+
+	if(status==0)
+	{
+		return handler.get_full_result();	
+	}
+
+	return std::string("failure-sax");
+}
+/*
+std::string run_filter_sax(const char * in, std::vector<std::string>& pattern)
+{
+	JsonParserHandler handler;
+
+
+	//handler.filter_pattern = {"glossary", "GlossDiv", "GlossList", "GlossEntry", "GlossDef", "postarray"};
+
+	handler.filter_pattern = pattern;
+
+	int status = handler.process_json_buffer(base64_decode(std::string(in)).data(), strlen(in));
+
+	if(status==0)
+	{
+		return handler.get_filter_result();	
+	}
+
+	return std::string("failure-sax");
+}*/
+
+std::string run_exact_filter(const char * in, std::vector<std::vector<std::string>>& pattern)
+{
+	JsonParserHandler handler;
+
+	int status{1};
+
+	handler.projection_exact.reserve(pattern[0].size() + pattern[1].size());
+	handler.projection_exact.insert(handler.projection_exact.end(), pattern[0].begin(), pattern[0].end());
+	handler.projection_exact.insert(handler.projection_exact.end(), pattern[1].begin(), pattern[1].end());
+   
+    if (pattern.size() == 3) {
+	handler.where_exact.reserve(pattern[0].size() + pattern[2].size());
+	handler.where_exact.insert(handler.where_exact.end(), pattern[0].begin(), pattern[0].end());
+	handler.where_exact.insert(handler.where_exact.end(), pattern[2].begin(), pattern[2].end());
+	}
+
```

Comment:
`projection_exact` and `where_exact` are for the same purpose, filter the correct key path.
there is no need for 2 data-member.

the data structure of these 2 variables is not correct.
the correct data structure should contain a dynamic number of key-paths.